source/framework/src/BookStoreManager.cc

```diff
@@ -0,0 +1,159 @@
+
+// -- marlin headers
+#include <marlin/BookStoreManager.h>
+#include <marlin/Application.h>
+#include <marlin/Exceptions.h>
+#include <marlin/Processor.h>
+
+#include <marlin/book/StoreWriter.h>  
+
+namespace marlin {
+
+#define INSTANCIATIONS_HIST(type) \
+  template std::optional<book::Handle<book::Entry<type>>> BookStoreManager::getObject<type>(\
+    const book::EntryKey*) const ;\
+                            \
+  template book::Handle<book::Entry<type>> BookStoreManager::bookHist<type>(\
+      const std::filesystem::path&,\
+      const std::string_view&,\
+      const std::string_view&,\
+      const std::array<\
+        const AxisConfig<typename type::Precision_t>*,\
+        type::Dimension>&,\
+      const BookFlag&) 
+
+  INSTANCIATIONS_HIST(Hist1F);
+  INSTANCIATIONS_HIST(Hist1D);
+  INSTANCIATIONS_HIST(Hist1I);
+  INSTANCIATIONS_HIST(Hist2F);
+  INSTANCIATIONS_HIST(Hist2D);
+  INSTANCIATIONS_HIST(Hist2I);
+  INSTANCIATIONS_HIST(Hist3F);
+  INSTANCIATIONS_HIST(Hist3D);
+  INSTANCIATIONS_HIST(Hist3I);
+
+  //--------------------------------------------------------------------------
+    
+  
+  BookStoreManager::~BookStoreManager() {
```

Comment:
Bad idea. Not good to write the book store content on destruction. Create a function to do it. Users should not be forced to write data to disk

Reply:
he isn't because it's only writing what is set to write.

But yes an extra function is more elegant :)

Reply:
>he isn't because it's only writing what is set to write.

Yes but we don't want a side effect of writing data to disk without asking explicitly for it.
Create a function but do not call it in the destructor. It should be empty for now